torch/csrc/lazy/core/ir.h

```diff
@@ -188,6 +193,43 @@ const T* NodeCast(const Node* node, OpKind op) {
 #endif
 }
 
+template <typename T, typename... Args>
+NodePtr ReuseNode(OpKind op, Args&&... args) {
+  if (FLAGS_torch_lazy_reuse_ir) {
+    std::deque<std::unique_ptr<TrieNode>>& successors =
+        Trie::Get()->Current()->successors;
+    for (auto it = successors.begin(); it != successors.end(); it++) {
+      NodePtr ir_node = (*it)->ir_node;
+      const T* concrete_node = NodeCast<T>(ir_node.get(), op);
+      if (concrete_node && concrete_node->Equal(std::forward<Args>(args)...)) {
+        TORCH_LAZY_COUNTER("IrNodeReused::" + std::string(typeid(T).name()), 1);
+        Trie::Get()->SetCurrent(it);
+        return ir_node;
+      }
+    }
+  }
+  return nullptr;
+}
+
+// TODO(alanwaketan): Support r-value reference argument type.
```

Comment:
@alanwaketan is this still worth fixing? is it hard to fix? performance of MakeNode is a little less critical if we land this PR but it would still be nice to do the 'best practices' thing if it's not too difficult